payment/src/main/java/org/killbill/billing/payment/api/DefaultPaymentGatewayApi.java

```diff
@@ -103,6 +105,14 @@ public GatewayNotification doPaymentGatewayApiOperation(final Iterable<PluginPro
                                             final CallContext callContext,
                                             final WithPaymentControlCallback<T> callback) throws PaymentApiException {
 
+        final RequestData requestData = Request.getPerThreadRequestData();
+        final String requestId;
+        if (requestData != null) {
+            requestId = requestData.getRequestId();
+        } else {
+            requestId = "NotAvailableRequestId";
+        }
+
```

Comment:
I agree. Therefore I moved the retrieval of the requestId to the places to the method where it is actually used and removed it here.